lib/cretonne/src/ir/funcname.rs

```diff
@@ -52,7 +52,7 @@ impl FunctionName {
 fn try_as_name(bytes: &[u8]) -> Option<String> {
     let mut name = String::with_capacity(bytes.len());
     for c in bytes.iter().map(|&b| b as char) {
-        if c.is_ascii() && c.is_alphanumeric() || c == '_' {
+        if c.is_ascii() && c.is_alphanumeric() || c == '_' || c == '.' || c == '-' {
```

Comment:
I don't think this would be good for the Cretonne IL syntax.

Cretonne's function names are used for linkage, but WebAssembly doesn't link functions by name — it uses numbers instead. So it is unlikely that we end up generating Cretonne functions with names from the WebAssembly name section anyway.

So there is no need for us to propagate the lispisms from WebAssembly's test suite.